lib/map.h

```diff
@@ -110,8 +111,8 @@ class IterKeys {
         }
         bool operator==(const iterator &i) const { return it == i.it; }
         bool operator!=(const iterator &i) const { return it != i.it; }
-        decltype(*&it->first) operator*() const { return it->first; }
-        decltype(&it->first) operator->() const { return &it->first; }
+        auto operator*() const { return it->first; }
+        auto operator->() const { return &it->first; }
```

Comment:
This is potentially inconsistent with the `pointer`/`reference` usings above. Also for `operator*` it derives a value instead of reference (as did the original code).